src/Result/Failure.php

```diff
@@ -19,18 +19,38 @@ class Failure implements FailureInterface
     private $reason;
 
     /**
+     * @var string|null
+     */
+    private $expectedOutput;
+
+    /**
+     * @var string|null
+     */
+    private $actualOutput;
+
+    /**
+     * @var string|null
+     */
+    private $errors;
+
+    /**
      * @var string
      */
     private $name;
 
     /**
      * @param string $name
      * @param string|null $reason
+     * @param string|null $expectedOutput
+     * @param string|null $actualOutput
      */
-    public function __construct($name, $reason = null)
+    public function __construct($name, $reason = null, $expectedOutput = null, $actualOutput = null, $errors = null)
```

Comment:
I don't think this class should contain output - this is more of a generic message - if there is problems with out the `StdOutFailure` should be used instead.